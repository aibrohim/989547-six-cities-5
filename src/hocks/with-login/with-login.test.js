import React from "react";
import ShallowRenderer from 'react-test-renderer/shallow';
import propTypes from "prop-types";

import {withLogin} from "./with-login";

const noop = () => {};

const MockComponent = (props) => {
  const {children} = props;

  return (
    <div>
      {children}
    </div>
  );
};

MockComponent.propTypes = {
  children: propTypes.oneOfType([
    propTypes.arrayOf(propTypes.node),
    propTypes.node
  ]).isRequired,
};

const MockComponentWrapped = withLogin(MockComponent);

it(`should with login render`, () => {
  const renderer = new ShallowRenderer();
  const tree = renderer
  .render(
      <MockComponentWrapped loginAction={noop}>
        <React.Fragment />
      </MockComponentWrapped>
  );

  expect(tree).toMatchSnapshot();
});
