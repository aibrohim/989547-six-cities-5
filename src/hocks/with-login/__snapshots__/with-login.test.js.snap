// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should with login render 1`] = `
<MockComponent
  buttonRef={
    Object {
      "current": null,
    }
  }
  isLogging={false}
  loginAction={[Function]}
  loginRef={
    Object {
      "current": null,
    }
  }
  onSubmit={[Function]}
  passwordRef={
    Object {
      "current": null,
    }
  }
>
  <React.Fragment />
</MockComponent>
`;
